### PR TITLE
pkg/trace/obfuscate: fix obfuscation error on SQL queries using the "!" operator

### PR DESCRIPTION
**Pull request by alexandre-normand**
### What does this PR do?

This fixes obfuscation errors on queries like `select * from table1 where !boolean_column` which is valid on mysql. 

To take an example:
```
# Given this table in mysql
CREATE table test(is_good boolean);

# This is valid but we fail to obfuscate
select * from test where !is_good;
```

Obfuscation error:
```
expected "=" after "!", got "i" (105)
```

### Possible Drawbacks / Trade-offs

It's a little tricky to cover all edge-cases. For example, `-` is an invalid character after `!` in a where clause but is valid if it's in the select columns:

```
select * from test where is_good !- 2;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '! -2' at line 1

mysql> select !- 2;
+------+
| !- 2 |
+------+
|    0 |
+------+
```

Therefore, the attempt here is to preserve obfuscation while minimizing false cases of errors. This means that the obfuscator would not mind that `select * from test where is_good !- 2` is not a valid query. 

### Describe how to test/QA your changes

Unit tests provide good coverage. 

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The appropriate `team/..` label has been applied, if known.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/trace/obfuscate/sql_tokenizer.go [hunk 1]
```diff
@@ -296,11 +298,11 @@ func (tkn *SQLTokenizer) Scan() (TokenKind, []byte) {
 			}
 			return TokenKind(ch), tkn.bytes()
 		case '!':
-			switch tkn.lastChar {
-			case '=':
+			switch c := tkn.lastChar; {
+			case c == '=':
```
Comment (gbbr) [inline pkg/trace/obfuscate/sql_tokenizer.go:302]: I assume this change and the one below was unintentionally committed.
```suggestion
			switch tkn.lastChar; {
			case '=':
```

Comment (alexandre-normand) [reply pkg/trace/obfuscate/sql_tokenizer.go]: It wasn't really unintentional. I needed to change the switch style to call `isValidCharAfterOperator(c)` but I revisited that implementation style after seeing how we just put those in the `default` case in other places in the tokenizer. I updated the PR with that version.

File: pkg/trace/obfuscate/sql_tokenizer.go [hunk 2]
```diff
@@ -310,7 +312,11 @@ func (tkn *SQLTokenizer) Scan() (TokenKind, []byte) {
 					return NE, []byte("!~")
 				}
 			default:
-				tkn.setErr(`expected "=" after "!", got "%c" (%d)`, tkn.lastChar, tkn.lastChar)
+				if isValidCharAfterOperator(tkn.lastChar) {
+					return Not, tkn.bytes()
+				}
+
```
Comment (gbbr) [inline pkg/trace/obfuscate/sql_tokenizer.go:318]: Can you please remove this empty line? I should maybe link to our [style guide](https://github.com/DataDog/dd-trace-go/wiki/Style-guidelines#general-guidelines) (point 3) somewhere in this repo too. Until then, I'll have to annoy you with my comments.

**Review by gbbr — APPROVED** [review]
Ah, this is much nicer 🙂 Thanks a lot!

File: pkg/trace/obfuscate/sql_tokenizer.go (hunk 1 above)
Comment (gbbr) [reply pkg/trace/obfuscate/sql_tokenizer.go]: Of course 🤦🏻 Thanks for explaining!

File: pkg/trace/obfuscate/sql_tokenizer.go (hunk 2 above)
Comment (alexandre-normand) [reply pkg/trace/obfuscate/sql_tokenizer.go]: 😁 . Of course, blank line removed ✅ .

**Review by dujuku — APPROVED** [review]
LGTM 👍
